tests/test_binning.py

```diff
@@ -334,4 +340,186 @@ def test_init(self):
 
     def test_determine_total_bins(self):
         mab = MABBinMapper([5])
+        # Test bin counting
         assert mab.determine_total_bins(nbins_per_dim=[5, 1], direction=[1, 86], skip=[0, 0], bottleneck=True) == 9
+        assert mab.determine_total_bins(nbins_per_dim=[5, 5], direction=[0, 0], skip=[0, 0], bottleneck=True) == 33
+        assert mab.determine_total_bins(nbins_per_dim=[5, 5, 5], direction=[0, 0, 0], skip=[0, 0, 0], bottleneck=True) == 137
+        assert mab.determine_total_bins(nbins_per_dim=[5, 5], direction=[0, 0], skip=[1, 0], bottleneck=True) == 9
+        assert mab.determine_total_bins(nbins_per_dim=[5, 1], direction=[1, 86], skip=[0, 0], bottleneck=False) == 6
+
+        # Test bin assignments with 2D and 3D coords
+        # First load reference datasets from file
+        test_refs = []
+        with h5py.File(os.path.join(REFERENCE_PATH, 'mab_assignments_ref.h5'), 'r') as f:
+            for i in range(11):
+                test_refs.append(f[f'test_result_{i:d}'][:])
+
+        # Create the same synthetic test data used in reference dataset
+        # The data is a 2D array of grid points
+        N_point = 15
+        n_dim = 2
+        N_total = N_point**n_dim
+        coords = np.zeros((N_total, 2))
+        for i in range(N_point**n_dim):
+            coords[i, 0] = i % int(N_point)
+            coords[i, 1] = i // int(N_point)
+        coords /= N_point
+
+        # generate weights as a n_dim gaussian where the max is at the center of the 2D space
+        weights = np.zeros(N_total)
+        for i in range(N_total):
+            weights[i] = np.exp(-((coords[i, 0] - N_point / 2) ** 2 + (coords[i, 1] - N_point / 2) ** 2) / (N_point / 2) ** 2)
+        weights /= np.sum(weights)
+        allcoords = np.ones((N_total, 4))
+        allcoords[:, :2] = coords
+        allcoords[:, 2] = weights
+        allcoords = np.tile(allcoords, (2, 1))
+        allcoords[0:N_total, 3] = 0
+        mask = np.full((N_total * 2), True)
+        output = list(np.zeros((N_total * 2), dtype=int))
+
+        # TEST 1: one direction only, no bottleneck
```

Comment:
Might be easier to separate all these tests into separate methods. That way it'll be easier to track which one is failing (and/or if all of them are failing). 

The reading of the reference h5file, prepping the fake dataset weights/coords from a Gaussian distribution could probably be moved into a fixture (https://github.com/westpa/westpa/blob/westpa2/tests/conftest.py).  In fact it might be cleaner to use pytest parametrize to consolidate most of the tests (https://github.com/westpa/westpa/blob/westpa2/tests/test_tools/test_w_fluxanl.py#L19-L23)

Reply:
Will look into rearranging the tests when I have more time, but they are at least all passing at the moment.

Reply:
I've refactored the tests as per your suggestions.